portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -70,7 +62,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query(COMMENT_ENTITY);
     PreparedQuery results = datastore.prepare(query);
-    List<Entity> messages = results.asList(FetchOptions.Builder.withLimit(numComments));
+
+    // Process user-selected maximum number of comments.
+    String numCommentsStr = request.getParameter(MAX_PARAMETER);
+    int numComments;
+    try {
+      numComments = Integer.parseInt(numCommentsStr);
+    } catch (Exception e) {
+      numComments = -1;
+    }
+
+    List<Entity> messages = (numComments == -1) ? messages = results.asList(FetchOptions.Builder.withDefaults())
+      : results.asList(FetchOptions.Builder.withLimit(numComments));
```

Comment:
Here's a suggestion on how to make this more readable:

```suggestion
    FetchOptions fetchOptions = numComments > 0 ? FetchOptions.Builder.withLimit(numComments) : FetchOptions.Builder.withDefaults();
    List<Entity> messages = results.asList(fetchOptions);
```